qt/widgets/instrumentview/test/PlotFitAnalysisPanePresenterTest.h

```diff
@@ -0,0 +1,95 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+
+#include "MantidQtWidgets/InstrumentView/PlotFitAnalysisPaneMocks.h"
+#include "MantidQtWidgets/InstrumentView/PlotFitAnalysisPanePresenter.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+#include <string>
+#include <utility>
+
+using namespace Mantid::API;
+using namespace testing;
+using Mantid::Geometry::Instrument;
+using namespace MantidQt::MantidWidgets;
+
+class PlotFitAnalysisPanePresenterTest : public CxxTest::TestSuite {
+public:
+  PlotFitAnalysisPanePresenterTest() { FrameworkManager::Instance(); }
+
+  static PlotFitAnalysisPanePresenterTest *createSuite() {
+    return new PlotFitAnalysisPanePresenterTest();
+  }
+
+  static void destroySuite(PlotFitAnalysisPanePresenterTest *suite) {
+    delete suite;
+  }
+
+  void setUp() override {
+    m_view = new NiceMock<MockPlotFitAnalysisPaneView>();
+    m_model = new MockPlotFitAnalysisPaneModel();
+    m_presenter = new PlotFitAnalysisPanePresenter(m_view, m_model);
+  }
+
+  void tearDown() override {
+    AnalysisDataService::Instance().clear();
+    delete m_view;
+    delete m_presenter;
```

Comment:
Should the model also be deleted here?

Reply:
the model should be deleted as part of the presenter destructor. Adding it as `delete m_model` causes a seg fault. So I have just made it equal to `nullptr`